src/gc/gc.d

```diff
@@ -2122,6 +2308,12 @@ struct Gcx
                 auto pool = findPool(p);
                 if (pool)
                 {
+                    static if (precise) if (p1pool)
+                    {
+                        size_t p1off = cast(size_t)(p1 - cast(void**)p1pool.baseAddr);
```

Comment:
The previous versions had some optimization to this respect, but I wanted to make minimal changes to begin with. The benchmarks don't show a large increase in scanning time, the preformance decrease is caused by setting the pointer bitmaps.